cmd/rocsp-tool/client.go

```diff
@@ -37,25 +37,36 @@ type processResult struct {
 	err error
 }
 
-func (cl *client) loadFromDB(ctx context.Context, speed ProcessingSpeed) error {
+func getStartingID(ctx context.Context, clk clock.Clock, db *sql.DB) (int64, error) {
 	// To scan the DB efficiently, we want to select only currently-valid certificates. There's a
 	// handy expires index, but for selecting a large set of rows, using the primary key will be
 	// more efficient. So first we find a good id to start with, then scan from there. Note: since
 	// AUTO_INCREMENT can skip around a bit, we add padding to ensure we get all currently-valid
 	// certificates.
-	// TODO(#5783): Allow starting from a specific ID.
-	startTime := cl.clk.Now().Add(-24 * time.Hour)
+	startTime := clk.Now().Add(-24 * time.Hour)
 	var minID *int64
-	err := cl.db.QueryRowContext(
+	err := db.QueryRowContext(
 		ctx,
 		"SELECT MIN(id) FROM certificateStatus WHERE notAfter >= ?",
 		startTime,
 	).Scan(&minID)
 	if err != nil {
-		return fmt.Errorf("selecting minID: %w", err)
+		return 0, fmt.Errorf("selecting minID: %w", err)
 	}
 	if minID == nil {
-		return fmt.Errorf("no entries in certificateStatus (where notAfter >= %s)", startTime)
+		return 0, fmt.Errorf("no entries in certificateStatus (where notAfter >= %s)", startTime)
+	}
+	return *minID, nil
+}
+
+func (cl *client) loadFromDB(ctx context.Context, speed ProcessingSpeed, startFromID int64) error {
+	prevID := startFromID
```

Comment:
I don't think I see `startFromID` being used anywhere here, so you can just initialize it as `prevID`

```suggestion
func (cl *client) loadFromDB(ctx context.Context, speed ProcessingSpeed, prevID int64) error {
```

Reply:
I generally like to avoid modifying variables that came in as function parameters. I know it's _allowed_, but I find it confusing - if someone is reading far down in a function and sees that variable name, they might assume it has the value that was passed in, rather than some modified value.